save_embed.py

```diff
@@ -0,0 +1,70 @@
+import multiprocessing as mp
+import pickle as pk
+import sys
+from sentence_transformers import SentenceTransformer
+import sent2vec
+import numpy as np
+import pandas as pd
+
+model_weights_path_st = '../../data_files/distilbert-base-nli'
+model_weights_path_sv = '../../data_files/torontobooks_unigrams.bin'
+closed_d3m_file = '../../data_files/data/closed_d3m_data.csv'
+open_d3m_file = '../../data_files/data/open_d3m_data.csv'
+    
+def embed(df: pd.DataFrame, model_weights_path: str, embedding_model, use_col_name_only: bool):
```

Comment:
We still have two `embed` functions. Can we just have one? Maybe we could use a caching mechanism that stores the emebddings to file.

Reply:
Do you want me to just delete the d3m-profiler directory on my branch? All of my embedding and testing is done without it. That is where these comments are coming from. I do not actually need any of the d3m-profiler, everything that I need is in cross_validation_mpi.py and save_embed.py. I am not sure what you mean by a caching mechanism.

Reply:
We addressed the first part of this on Slack: edit the original code with your code.

What I mean by a caching mechanism is just some code with the following logic:
```
def embed(X, model):
  id = deterministic_hash_of_some_kind(X, model)
  embedding_file_path = './embeddings/{}.ext'.format(id)
  if os.path.isfile(embedding_file_path):
    with open(embedding_file_path, 'r') as f:
      embeddings = f.read() 
  else:
    embeddings = model.encode(X)
    with open(embedding_file_path, 'w') as f:
      f.write(embeddings)
  return embeddings
```

Reply:
It seems to me like the code in this file has made it into `d3m_profiler/embed.py` and this file needs to be deleted now.